Dockerfile

```diff
@@ -8,6 +9,7 @@ WORKDIR /workspace
 COPY . .
 
 RUN adduser -D -u 10001 xray
+ENV TARGETPLATFORM=${TARGETPLATFORM:-linux/amd64}
```

Comment:
I'm also still confused why the environment variable always has the `amd64` suffix, when we're also building for ARM?

Reply:
I was also confused about this value but then I tested this in my docker file. I think when this script will be called twice in the case of `linux/amd64` and `linux/arm64`. So in the case of `linux/arm64` value of `TARGETPLATFORM` will be `linux/arm64` only. I would consider this line as a fallback assignment (probably if someone is running dockerfile without giving any platform).

Reply:
Ahh ok so the `:-` is an [operator](https://stackoverflow.com/questions/10390406/usage-of-colon-dash-in-bash) to fall back on amd64 if TARGETPLATFORM is undefined. Gotta love bash.